proposals/lip-0035.md

```diff
@@ -64,25 +64,27 @@ The state of an account on the Lisk mainnet after applying the block at height `
 * The array `sentVotes` in the account asset is set to the empty array.
 * The array `unlocking` in the account asset is set to the empty array.
 
-In order to ensure the uniqueness of the account state, we perform the following additional checks:
+In order to ensure the uniqueness of the account state and handle non-unique string representations of Lisk addresses in the past, we additionally perform the following steps:
 
-* Ensure that an account has a 20-byte address if there is at least one outgoing transaction from that account and an 8-byte address otherwise.
-* Ensure that there is no account with no incoming transaction.
-* Ensure that there is at most one account for every 8-byte binary address (legacy address format). This means that if two accounts have different string addresses corresponding to the same 8-byte address (e.g., `123L` and `0123L`), they are merged by adding the respective balances.
+* Discard any account without incoming transaction (such accounts cannot have any balance).
+* Discard the so-called *Genesis Account* with public key `d121d3abf5425fdc0f161d9ddb32f89b7750b4bdb0bff7d18b191d4b4bafa6d4` (Lisk address `6566229458323231555L` in the old format), which has negative balance.
+* Migrate every account, except the Genesis Account and any account without incoming transaction, as follows.
+  * If the string representation of an account in Lisk Core 2.0 is standard (The string is `0L` or does not start with `0`. The decimal number is between `0` and `2^64-1` inclusive), then the account is migrated as defined above. If the account has at least one outgoing transaction, then the migrated account has a 20-byte address and otherwise it has an 8-byte address.
+  * If the string representation of an account in Lisk Core 2.0 is non-standard (It is not `OL`, but starts with `0`. Or the decimal number is larger than `2^64-1`), the account corresponds to one unique 8-byte address, denoted by `legacyAddress`, which are the 8 bytes signed in the transaction creating the account. Note that the account cannot have any outgoing transaction as the string representation is non-standard. The non-standard account `acc_non_standard` is migrated as follows:
```

Comment:
I find this point a bit confusing: 
Regarding the first sentence: 
a) there can be more than one non-standard accounts with the same 8-byte address, as it is said in the last sentence of the line 75, right?
b) did not the transaction creating the account sign the decimal + L string and that is why we have addresses over the 8 bytes range?

Reply:
I  believe the specs are right, but the way it is written confuses me. the Rationale section clarified it afterwards.

Reply:
In the Lisk Core 2.0 serialization, the string representation (`123L`) is serialized as 8-bytes, see https://github.com/LiskHQ/lisk-sdk/blob/v2/elements/lisk-transactions/src/utils/get_transaction_bytes.ts.